src/enclave/http.h

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "tlsendpoint.h"
+
+#include <http-parser/http_parser.h>
+
+namespace enclave
+{
+  namespace http
+  {
+    std::vector<uint8_t> post(const std::string& body)
+    {
+      auto req = fmt::format(
+        "POST / HTTP/1.1\r\n"
+        "Content-Type: application/json\r\n"
+        "Content-Length: {}\r\n\r\n{}",
+        body.size(),
+        body);
+      return std::vector<uint8_t>(req.begin(), req.end());
+    }
+
+    std::vector<uint8_t> post_header(const std::vector<uint8_t>& body)
+    {
+      auto req = fmt::format(
+        "POST / HTTP/1.1\r\n"
```

Comment:
We should define this header string only once.

Reply:
Yes, we'll probably end up a non-trivial request formatter soon.